app/controllers/corrections/CorrectPreviousReturnController.scala

```diff
@@ -56,10 +60,19 @@ class CorrectPreviousReturnController @Inject()(
           Future.successful(BadRequest(view(formWithErrors, mode, period))),
 
         value =>
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectPreviousReturnPage, value))
-            _              <- cc.sessionRepository.set(updatedAnswers)
-          } yield Redirect(CorrectPreviousReturnPage.navigate(mode, updatedAnswers))
-      )
+      for {
+        updatedAnswers <- Future.fromTry(request.userAnswers.set(CorrectPreviousReturnPage, value))
+        periods <- returnStatusConnector.listStatuses(request.registration.commencementDate)
```

Comment:
Can we move the connector call to inside the yield as if it fails then our session repo wont get the updatedUserAnswers

Reply:
OR just reorder them so periods is the last call would work

Reply:
done, reordered them